api/app/src/command/webhook/webhook.ts

```diff
@@ -200,24 +214,60 @@ const processOneCustomer = async (
   return true;
 };
 
+export const processPatientRequest = async (
+  cxId: string,
+  patientId: string,
+  documents: DocumentReferenceDTO[]
+): Promise<boolean> => {
+  try {
+    const settings = await getSettingsOrFail({ id: cxId });
+    // create a representation of this request and store on the DB
+    const payload: WebhookPatientDataPayloadWithoutMessageId = {
+      patients: [{ patientId, documents }],
+    };
+    const webhookRequest = await createWebhookRequest({ cxId, payload });
+    // send it to the customer and update the request status
+    await processRequest(webhookRequest, settings, ApiTypes.medical);
+
+    // TODO: #484
+    // reportUsage(cxId, ...);
+  } catch (err) {
+    log(`Error on processPatientRequest: `, err);
+    capture.error(err, {
+      extra: { patientId, context: `webhook.processPatientRequest` },
+    });
+  }
+  return true;
+};
+
 export const processRequest = async (
   webhookRequest: WebhookRequest,
-  settings: Settings
+  settings: Settings,
+  apiType: ApiTypes = ApiTypes.devices
 ): Promise<boolean> => {
   const payload = webhookRequest.payload;
 
   const { webhookUrl, webhookKey, webhookEnabled } = settings;
   if (!webhookUrl || !webhookKey) {
-    console.log(
+    const msg =
       `Missing webhook config, skipping sending it ` +
-        `(url: ${webhookUrl}, key: ${webhookKey ? "<defined>" : null}`
-    );
-    // mark this request as failed on the DB - so it can be retried later
-    await updateWebhookRequestStatus({
-      id: webhookRequest.id,
-      status: "failure",
-    });
-    return false;
+      `(url: ${webhookUrl}, key: ${webhookKey ? "<defined>" : null}`;
+    console.log(msg);
+    capture.message(msg, { extra: { context: `webhook.processRequest` } });
+    // if this is for DAPI:
+    //    mark this request as failed on the DB - so it can be retried later
+    // if this is for MAPI:
+    //    silently ignore this since this is just a notification for ease-of-use
+    //    and won't result in data loss
```

Comment:
Is there a benefit of not enabling retries for doc references? Just trying to gauge the cost/effort of doing it vs the added complexity to the API by having diff behaviors based on the API the customer is using.

If we keep like this, I think we should document it on the retry section of webhook docs.

Reply:
^ to be explicit, the retries will still function as expected if the cx has the WH url setup

Reply:
> if they don't have it set, that means they don't have the endpoint implemented yet
One might delete the API after setting it up.

> there should be no benefit of retrying these payloads after they implement the WH, as they will be able to get the same data using the mechanism they previously used, and only new data will be sent
That assumes they would implement `GET /document`, but _in theory_ customers could only implement webhook, as it sends all the document references we have for a given patient, right?

Reply:
As discussed offline:

- We will retry doc refs, if the failure reason is for something other than them not having a WH url set on the dash. The logic here is that if they don't initially (let's say they choose not to implement WH in their flow), when they turn the switch on, they'll get a bunch of old doc refs that they should have already processed, so they'll have to filter those out or something - trying to avoid those situations.
- Good point as to what if they change/remove the URL after it has already been setup once though - will document this limitation for now and iterate based on cx feedback